src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -947,6 +947,49 @@ public static class Extensions
         Assert.Equal("Extensions.<>E__1<T>", symbol2.ToTestDisplayString());
     }
 
+    [Fact]
+    [WorkItem("https://github.com/dotnet/roslyn/issues/78135")]
+    public void ExtensionWithCapturing()
+    {
+        var src = """
+using System;
+using System.Text;
+
+var sb = new StringBuilder("Info: ");
+Extensions.Inspect(sb);
+
+public static class Extensions
+{
+    extension(StringBuilder)
+    {
+        public static StringBuilder Inspect(StringBuilder sb)
+        {
+            var s = () =>
+            {
+                foreach (char c in sb.ToString())
+                {
+                    Console.Write(c);
+                }
+            };
+            s();
+            return sb;
+        }
+    }
+}
+""";
+
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
```

Comment:
>comp.VerifyEmitDiagnostics(); [](http://example.com/codeflow?start=8&length=29)

I think we would want to execute the code and observe that it works as expected.

Reply:
Yeah that would be great - not sure how.
And we should may at a test where we can see that only captured locals are are hoisted to fields in display classes.

e.g. in code like:

```csharp
using System;
using System.Text;

int.Foo();

public static class Test
{
    extension(int)
    {
        public static void Foo()
        {
            var b = 123;
            Action a = () =>
            {
                int a = 123;
                Console.WriteLine(a);
                Console.WriteLine(b);
            };
        }
    }    
}
```

the compiler should generate only a field for `b` but not for `a`

Reply:
```ComileAndVerify``` helper can be used to assert expected execution output.

> And we should may add a test where we can see that only captured locals are are hoisted to fields in display classes.

```VerifyTypeIL``` helper might help with that. However, when optimization level is DEBUG, all locals might be hoisted by design.